externals/kyuubi-jdbc-engine/src/test/scala/org/apache/kyuubi/engine/jdbc/starrocks/StarRocksOperationWithEngineSuite.scala

```diff
@@ -75,4 +75,26 @@ class StarRocksOperationWithEngineSuite extends StarRocksOperationSuite with Hiv
       assert(tFetchResultsResp.getStatus.getStatusCode === TStatusCode.SUCCESS_STATUS)
     }
   }
+
+  test("starrocks - JDBC ExecuteStatement operation cancel should kill SQL statement") {
+    withSessionHandle { (client, handle) =>
+      val tExecuteStatementReq = new TExecuteStatementReq()
+      tExecuteStatementReq.setSessionHandle(handle)
+      tExecuteStatementReq.setStatement("SELECT sleep(1200)")
+      tExecuteStatementReq.setRunAsync(true)
+      val tExecuteStatementResp = client.ExecuteStatement(tExecuteStatementReq)
+
+      // todo:we need to improvement the cancel mechanism in starrocks server side,
+      //  once a query submitted, we should be able to cancel it immediately.
```

Comment:
it's unclear to me what the current behavior is from the words, and if you have StarRocks side issues/PR links, please leave them here.

Reply:
Sorry, I didn't express myself clearly. I moved this `todo` to `ExecuteStatement`. 
This is an issue present in all JDBC engine in kyuubi. When using asyn mode, the statement creation is delayed. If we cancel immediately after submitting the SQL, there is a chance the cancellation may fail. If the statement has not yet been created at that point,

Reply:
ah, that makes sense